samples/caf_sensor_manager/src/modules/sensor_stub_gen.c

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/kernel.h>
+#include <drivers/sensor_stub.h>
+
+#define MODULE sensor_stub_gen
+#include <caf/events/module_state_event.h>
+
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(MODULE, CONFIG_SENSOR_STUB_GEN_LOG_LEVEL);
+
+
+#define SENSOR_FRACTIONAL_MAX (1000000)
+#define SENSOR_STUB_GEN_X_STEP (0.1)
+#define SENSOR_STUB_GEN_X_MAX  (5.0)
+#define SENSOR_STUB_GEN_X_MIN  (-5.0)
+
+#define SENSOR_STUB_GEN_Y_STEP (0.1)
+#define SENSOR_STUB_GEN_Y_MAX  (2.0)
+#define SENSOR_STUB_GEN_Y_MIN  (-2.0)
+
+#define SENSOR_STUB_GEN_Z_STEP (0.2)
+#define SENSOR_STUB_GEN_Z_MAX  (1.0)
+#define SENSOR_STUB_GEN_Z_MIN  (0.0)
+
+#define SENSOR_STUB_SENSOR_FP(x) ((int64_t)((x) * SENSOR_FRACTIONAL_MAX))
+
+enum {
+	SENSOR_STUB_CH_X,
+	SENSOR_STUB_CH_Y,
+	SENSOR_STUB_CH_Z,
+	SENSOR_STUB_CH_CNT,
+};
+
+/* All the data for implemented sensor simulation */
+struct sensor_stub_gen_ch_data {
+	/* Current value */
+	struct sensor_value val;
+	/* Current direction of the changes */
+	bool dir;
+};
+
+/* Parameters of the signal generated on the selected channel */
+struct sensor_stub_gen_ch_params {
+	int64_t step, max, min;
+};
+
+/* SENSOR_CHAN_ACCEL_X, SENSOR_CHAN_ACCEL_Y, SENSOR_CHAN_ACCEL_Z */
+static struct sensor_stub_gen_ch_data gen_data[SENSOR_STUB_CH_CNT];
+
+static const struct sensor_stub_gen_ch_params gen_params[SENSOR_STUB_CH_CNT] = {
+	[SENSOR_STUB_CH_X] = {
+		.step = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_X_STEP),
+		.min  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_X_MIN),
+		.max  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_X_MAX)
+	},
+	[SENSOR_STUB_CH_Y] = {
+		.step = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Y_STEP),
+		.min  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Y_MIN),
+		.max  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Y_MAX)
+	},
+	[SENSOR_STUB_CH_Z] = {
+		.step = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Z_STEP),
+		.min  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Z_MIN),
+		.max  = SENSOR_STUB_SENSOR_FP(SENSOR_STUB_GEN_Z_MAX)
+	}
+};
+
+static inline int64_t sensor_val_2_fp(const struct sensor_value *val)
+{
+	return ((int64_t)val->val1 * SENSOR_FRACTIONAL_MAX) + val->val2;
+}
+
+static inline void fp_2_sensor_val(struct sensor_value *out, int64_t val)
+{
+	out->val1 = (int32_t)(val / SENSOR_FRACTIONAL_MAX);
+	out->val2 = (int32_t)(val % SENSOR_FRACTIONAL_MAX);
+}
+
+/* Generate changes on selected channel with given parameters */
+static inline void sensor_stub_gen_chan(struct sensor_stub_gen_ch_data *ch_data,
```

Comment:
Maybe sensor_stub_gen_changes - current name can be easily confused chan -> channel

Reply:
Well - this was exactly my idea - the function is Generate Channel, not Generate Changes. I do not think this is relative anyway. It could be even named gen_data... I do not have strong opinion here.

Reply:
It's just a suggestion if you think it is ok leave it as it is.